### ISSUE TITLE
DVM Dataset Processing Out of Memory Issue

### ISSUE THREAD

**Issue by lyrain2001** [paulhager/MMCL-Tabular-Imaging#14]
Hi,

Firstly, I'd like to express my admiration for the impressive work. 

I encountered an Out Of Memory issue when running create_dvm_dataset.ipynb, particularly at the image normalization and saving step. To align my setup with yours and ensure a smooth run, could you please share the GPU memory specifications and any batch size or optimization strategies used in your environment?

Thanks in advance for your time.

**Comment by paulhager** [comment]
Image normalization should not be running on GPU, so GPU memory is not relevant. This is running on your CPU and using RAM. If you run out of RAM while preprocessing I recommend either saving the dataset in smaller increments or using live loading during training which means you can skip the preprocessing step entirely.

**Comment by lyrain2001** [comment]
My problem has been solved. Thanks for your help!

**Comment by dhshin1125** [comment]
Hello, I found some problem and this post
fortunatly, I solved problem
The problem was "Save Normalized Ims" cell
you use memory but not free inside the cell
so after that cell, it was crash

solution is simple, add del keyword for big local variable
original code
```
if front_view_only:
  IMAGES = join(BASE, 'Confirmed_fronts')
else:
  IMAGES = join(BASE, 'resized_DVM')

for df, t_split in zip([train_df, val_df, test_df], ['train', 'val', 'test']):
  images = []
  for i,row in df.iterrows():
    image_name = row['Image_name']
    split = image_name.split('$$')
    
    if front_view_only:
      path = join(IMAGES,split[0],split[2],image_name)
    else:
      path = join(IMAGES,split[0],split[1],split[2],split[3],image_name)
    images.append(read_image(path))
  images_t = torch.stack(images).float()
  images_t = images_t/255
  check_or_save(images_t, join(FEATURES, f'{t_split}_images{addendum}.pt'))
```

solution
```
if front_view_only:
  IMAGES = join(BASE, 'Confirmed_fronts')
else:
  IMAGES = join(BASE, 'resized_DVM')

for df, t_split in zip([train_df, val_df, test_df], ['train', 'val', 'test']):
  images = []
  for i,row in df.iterrows():
    image_name = row['Image_name']
    split = image_name.split('$$')
    
    if front_view_only:
      path = join(IMAGES,split[0],split[2],image_name)
    else:
      path = join(IMAGES,split[0],split[1],split[2],split[3],image_name)
    images.append(read_image(path))
  images_t = torch.stack(images).float()
  images_t = images_t/255
  check_or_save(images_t, join(FEATURES, f'{t_split}_images{addendum}.pt'))
  del df
  del t_split
  del images
  del images_t
```
